src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -546,13 +551,12 @@ public int MaxResponseDrainSize
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && !_disposed)
+            if (!_disposed)
```

Comment:
For WinHttpHandler, SYNC pattern is used.  That implies that if a callback is registered for the `requestHandle`, then the callback will be run sync in the same thread and within the actual WinHttpCloseHandle method.

And for the current implementation of WinHttpHandler, the callback is de-registered after the request/response cycle is done.  We don't use it (yet) for the WinHttpResponseStream.